vividus-plugin-aws-s3/src/test/java/org/vividus/aws/s3/steps/S3BucketStepsTests.java

```diff
@@ -64,50 +60,47 @@
     private S3BucketSteps steps;
 
     @Test
-    public void uploadFileTest() throws Exception
+    void uploadFileTest()
     {
+        byte[] csv = ResourceUtils.loadResourceAsByteArray(CSV_FILE_PATH);
         String contentType = "contentType";
-        byte[] resource = "string".getBytes(StandardCharsets.UTF_8);
-
-        ObjectMetadata objectMetadata = Mockito.mock(ObjectMetadata.class);
-        ByteArrayInputStream inputStream = Mockito.mock(ByteArrayInputStream.class);
-        PowerMockito.mockStatic(ResourceUtils.class);
-        PowerMockito.when(ResourceUtils.loadResourceAsByteArray(CSV_FILE_PATH)).thenReturn(resource);
-        PowerMockito.whenNew(ObjectMetadata.class).withNoArguments().thenReturn(objectMetadata);
-        PowerMockito.whenNew(ByteArrayInputStream.class).withArguments(resource).thenReturn(inputStream);
 
         steps.uploadResource(CSV_FILE_PATH, S3_OBJECT_KEY, contentType, S3_BUCKET_NAME);
-        verify(objectMetadata).setContentType(contentType);
-        verify(objectMetadata).setContentLength(resource.length);
-        verify(amazonS3Client).putObject(S3_BUCKET_NAME, S3_OBJECT_KEY, new ByteArrayInputStream(resource),
-                objectMetadata);
+        verify(amazonS3Client).putObject(eq(S3_BUCKET_NAME), eq(S3_OBJECT_KEY), argThat(bais -> {
+            try
+            {
+                byte[] actual = IOUtils.toByteArray(bais);
+                bais.reset();
+                return Arrays.equals(csv, actual);
+            }
+            catch (IOException e)
+            {
+                return false;
```

Comment:
why not to re-throw it? if the exception really occurs it would be hard to understand what is the reason the matching fails

Reply:
in current case the exception will never be thrown, but I need to catch checked exceptions